src/main/java/com/myorg/mvc/service/UserServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package com.myorg.mvc.service;
+
+import com.myorg.mvc.entity.User;
+import com.myorg.mvc.entity.UserDetailsImpl;
+import com.myorg.mvc.exceptions.DoubleRegistrationException;
+import com.myorg.mvc.exceptions.UserNotFoundException;
+import com.myorg.mvc.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@Service("UserService")
+public class UserServiceImpl implements UserService, UserDetailsService {
+    private static final String REGEX_PATTERN = "^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,7}$";
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) {
+        User user = userRepository.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("User not found"));
+        return new UserDetailsImpl(user);
+    }
+
+    @Override
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public User getUserById(Long id) {
```

Comment:
Should method have @Override annatation or not? half your methods have annotation, other half have not.